src/GitObjectDb/Models/Merge/ObjectRepositoryMerge.cs

```diff
@@ -20,57 +19,41 @@ namespace GitObjectDb.Models.Merge
 {
     /// <inheritdoc/>
     [ExcludeFromGuardForNull]
-    public sealed class ObjectRepositoryMerge : IObjectRepositoryMerge
+    internal sealed class ObjectRepositoryMerge : IObjectRepositoryMerge
     {
-        private readonly IServiceProvider _serviceProvider;
         private readonly IRepositoryProvider _repositoryProvider;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove the field '_repositoryProvider' and declare it as a local variable in the relevant methods. ([csharpsquid:S1450](https://sonarcloud.io/organizations/frblondin-github/rules#rule_key=csharpsquid%3AS1450))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=GitObjectDb&pullRequest=33&issues=AWe4j_Q1NP9b202rIM5L&open=AWe4j_Q1NP9b202rIM5L)

[ISSUE_KEY]: AWe4j_Q1NP9b202rIM5L